shared_memory_dict/dict.py

```diff
@@ -54,15 +74,71 @@ def __delitem__(self, key: str) -> None:
         with self._modify_db() as db:
             del db[key]
 
+    def __iter__(self) -> Iterator:
+        return iter(self._read_memory())
+
+    def __reversed__(self):
+        return reversed(self._read_memory())
+
     def __del__(self) -> None:
         self.cleanup()
 
-    def __contains__(self, key: object) -> bool:
+    def __contains__(self, key: str) -> bool:
         return key in self._read_memory()
 
+    def __eq__(self, other: Any) -> bool:
+        return self._read_memory() == other
+
+    def __ne__(self, other: Any) -> bool:
+        return self._read_memory() != other
+
+    if sys.version_info > (3, 8):
+
+        def __or__(self, other: Any):
+            return self._read_memory() | other
+
+        def __ror__(self, other: Any):
+            return other | self._read_memory()
+
+        def __ior__(self, other: Any):
+            with self._modify_db() as db:
+                db |= other
+                return db
+
+    def __str__(self) -> str:
+        return str(self._read_memory())
+
+    def __repr__(self) -> str:
+        return repr(self._read_memory())
+
+    def get(self, key: str, default: Optional[Any] = None) -> Any:
+        return self._read_memory().get(key, default)
+
     def keys(self) -> KeysView[Any]:  # type: ignore
         return self._read_memory().keys()
 
+    def values(self) -> ValuesView[Any]:  # type: ignore
+        return self._read_memory().values()
+
+    def items(self) -> ItemsView:  # type: ignore
+        return self._read_memory().items()
+
+    __SENTINEL = object()
```

Comment:
What about moving this sentinel to the top of the file and make it public?

Reply:
done!